stdlib/std.jsonnet

```diff
@@ -88,6 +88,9 @@ limitations under the License.
     range(from, to)::
         std.makeArray(to - from + 1, function(i) i + from),
 
+    slice(arr, from, to)::
+        std.makeArray(to - from, function(i) arr[i + from]),
```

Comment:
This should also work on strings and fail explicitly if arr is not a string or array type.  There is already str.substr but its semantics aren't right (see comment below).  So we should deprecate str.substr() and use this instead.